--- a/plugins/tray/CMakeLists.txt
+++ b/plugins/tray/CMakeLists.txt
@@ -21,7 +21,7 @@ find_package(Dtk6Widget REQUIRED)
 find_package(Dtk6Gui REQUIRED)
 find_package(Dtk6Core REQUIRED)
 find_package(DFrameworkdbusQt6 CONFIG REQUIRED)
-find_package(dbusmenu-lxqt CONFIG REQUIRED)
+find_package(dbusmenu-qt6 CONFIG REQUIRED)
 
 pkg_check_modules(XCB_LIBS REQUIRED xcb-ewmh xcb xcb-image xcb-composite xtst x11 xext xcb-icccm)
 pkg_check_modules(DDE-Network-Utils REQUIRED gxde-network-utils-qt6)
@@ -34,7 +34,7 @@ target_include_directories(${PLUGIN_NAME} PUBLIC ${DtkGui_INCLUDE_DIRS}
                                                  ${XCB_LIBS_INCLUDE_DIRS}
                                                  ${DDE-Network-Utils_INCLUDE_DIRS}
                                                  ${QGSettings_INCLUDE_DIRS}
-                                                 ${dbusmenu-lxqt_INCLUDE_DIRS}
+                                                 ${dbusmenu-qt6_INCLUDE_DIRS}
                                                  ../../interfaces
                                                  ../../frame)
 target_link_libraries(${PLUGIN_NAME} PRIVATE
@@ -47,7 +47,7 @@ target_link_libraries(${PLUGIN_NAME} PRIVATE
     ${XCB_LIBS_LIBRARIES}
     ${DDE-Network-Utils_LIBRARIES}
     #${dbusmenu-lxqt_LIBRARIES}
-    dbusmenu-lxqt
+    dbusmenu-qt6
     DFrameworkdbusQt6::DFrameworkdbusQt6
     ${QGSettings_LIBRARIES}
     pthread
--- a/plugins/tray/snitraywidget.cpp
+++ b/plugins/tray/snitraywidget.cpp
@@ -22,7 +22,7 @@
 #include "snitraywidget.h"
 #include "util/themeappicon.h"
 //#include <dbusmenu-qt6/dbusmenuimporter.h>
-#include <dbusmenu-lxqt/dbusmenuimporter.h>
+#include <dbusmenuimporter.h>
 #include <QPainter>
 #include <QApplication>
 
