--- a/plugins/dde-sys-monitor-plugin/CMakeLists.txt
+++ b/plugins/dde-sys-monitor-plugin/CMakeLists.txt
@@ -39,5 +39,6 @@ file(GLOB_RECURSE SRCS "*.h" "*.cpp" "*.ui")
 # <库名>_LIBRARIES      有哪些库文件（Qt5Widgets_LIBRARIES）
 find_package(Qt6 REQUIRED COMPONENTS Widgets)
+find_package(PkgConfig REQUIRED)
 pkg_check_modules(dtk2widget REQUIRED dtk2widget)
 
 # find_package 命令还可以用来加载 cmake 的功能模块
@@ -48,7 +49,6 @@ pkg_check_modules(dtk2widget REQUIRED dtk2widget)
 # PKG_CONFIG_FOUND            pkg-config 可执行文件是否找到了
 # PKG_CONFIG_EXECUTABLE       pkg-config 可执行文件的路径
 # PKG_CONFIG_VERSION_STRING   pkg-config 的版本信息
-find_package(PkgConfig REQUIRED)
 
 # 加载 FindPkgConfig 模块后就可以使用 pkg_check_modules 命令加载需要的库
 # pkg_check_modules 命令是由 FindPkgConfig 模块提供的，因此要使用这个命令必须先加载 FindPkgConfig 模块。
