--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,7 +27,7 @@ if (NOT (${CMAKE_BUILD_TYPE} MATCHES "Debug"))
 endif ()
 
 # Generate .qm files from .ts files
-find_program(QT_LRELEASE_EXECUTABLE /usr/lib/qt6/bin/lrelease)
+find_program(QT_LRELEASE_EXECUTABLE NAMES lrelease-qt6 lrelease HINTS /usr/lib64/qt6/bin REQUIRED)
 file(GLOB TS_FILES "${CMAKE_SOURCE_DIR}/translations/*.ts")
 set(QM_FILES)
 foreach(TS_FILE ${TS_FILES})
@@ -90,9 +90,9 @@ if (${EXIT_CODE} EQUAL 0)
     include(CPack)
 else()
     message(STATUS "apt not found! - CPank module disable")
-    install(FILES ${CMAKE_BINARY_DIR}/frame/dde-top-panel DESTINATION ${CMAKE_INSTALL_BINDIR} 
+    install(FILES ${CMAKE_BINARY_DIR}/frame/gxde-top-panel DESTINATION ${CMAKE_INSTALL_BINDIR}
         PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
-    install(FILES ${CMAKE_SOURCE_DIR}/xml/com.gxde.dde.toppanel.gschema.xml DESTINATION ${CMAKE_INSTALL_DATADIR}/glib-2.0/schemas)
+    install(FILES ${CMAKE_SOURCE_DIR}/xml/com.deepin.dde.toppanel.gschema.xml DESTINATION ${CMAKE_INSTALL_DATADIR}/glib-2.0/schemas)
-    install(FILES ${CMAKE_SOURCE_DIR}/dde-top-panel.desktop DESTINATION ${CMAKE_INSTALL_DATADIR}/applications)
+    install(FILES ${CMAKE_SOURCE_DIR}/gxde-top-panel.desktop DESTINATION ${CMAKE_INSTALL_DATADIR}/applications)
     install(DIRECTORY ${CMAKE_BINARY_DIR}/translations DESTINATION ${CMAKE_INSTALL_DATADIR}/gxde-top-panel)
 endif()
